test/test_open3.rb

```diff
@@ -284,4 +284,15 @@ def test_pipeline
     }
   end
 
+  def test_pipeline_err_to_out
+    cmd = [RUBY, '-e', 'STDERR.print "err"; STDOUT.print "out"']
+    cmd_expect_only_out = [RUBY, '-e', 's = STDIN.read; exit s == "out"']
+    cmd_expect_out_and_err = [RUBY, '-e', 's = STDIN.read; exit s == "errout"']
+    result_only_out = Open3.pipeline cmd, cmd_expect_only_out
+    result_out_and_err = Open3.pipeline cmd, cmd_expect_out_and_err, {:err => :out}
+    (result_only_out + result_out_and_err).each do |status|
+      assert status.success?
+    end
```

Comment:
`assert` for each results, like as:

```ruby
    command = [RUBY, '-e', 'STDERR.print "err"; STDERR.flush; STDOUT.print "out"']
    Open3.pipeline_r(command, err: %i[child out]) do |o|
      assert_equal "errout", o.read
    end
    command = [RUBY, '-e', 'STDERR.print "err"; STDERR.flush; STDOUT.print "out"']
    Open3.pipeline_r([*command, err: %i[child out]], [RUBY, '-pe;']) do |o|
      assert_equal "errout", o.read
    end
```

And, as above, you can do it with `[:child, :out]` already.

Reply:
For the last command which is not redirected, should use :err => :out.

```
% ruby -ropen3 <<'End' | cat -n
cmd1 = ['ruby', '-e', 'STDOUT.sync = true; STDOUT.puts "o1"; STDERR.puts "e1"', :err => %i[child out]]
cmd2 = ['ruby', '-e', 'STDOUT.sync = true; puts STDIN.read; STDOUT.puts "o2"; STDERR.puts "e2"', :err => %i[child out]]
cmd3 = ['ruby', '-e', 'STDOUT.sync = true; puts STDIN.read; STDOUT.puts "o3"; STDERR.puts "e3"', :err => %i[child out]]
cmd4 = ['ruby', '-e', 'STDOUT.sync = true; puts STDIN.read; STDOUT.puts "o4"; STDERR.puts "e4"', :err => :out]
Open3.pipeline(cmd1, cmd2, cmd3, cmd4)
End
     1	o1
     2	e1
     3	o2
     4	e2
     5	o3
     6	e3
     7	o4
     8	e4
```